src/reference/asciidoc/cassandra.adoc

```diff
@@ -0,0 +1,179 @@
+</int-cassandra:outbound-gateway>
+----
+====
+
+If `CassandraMessageHandler` is used as a gateway in an async mode, the `Mono<WriteResult>` is produced which is handled on reply producing according to the provided `MessageChannel` implementation.
+For true reactive processing a `FluxMessageChannel` is recommended for output channel configuration.
+In sync mode the `Mono.block()` is called for reply value.
+
+If  `INSERT`, `UPDATE` or `DELETE` operations are performed, the entity (marked `org.springframework.data.cassandra.core.mapping.Table`) is expected as a request message payload.
+If the payload is a list of entities, then respective batch operation is performed.
+
+The `ingestQuery` mode expects payload to be present as a matrix of values to insert - the `List<List<?>>`.
+For example, if the entity is like this:
+
+====
+[source,java]
+----
+@Table("book")
+public record Book(@PrimaryKey String isbn,
+                   String title,
+                   @Indexed String author,
+                   int pages,
+                   LocalDate saleDate,
+                   boolean isInStock) {
+
+}
+----
+====
+
+And channel adapter has this configuration:
+
+====
+[source,java]
+----
+@Bean
+public MessageHandler cassandraMessageHandler3() {
+    CassandraMessageHandler cassandraMessageHandler = new CassandraMessageHandler(this.template);
+    String cqlIngest = "insert into book (isbn, title, author, pages, saleDate, isInStock) values (?, ?, ?, ?, ?, ?)";
+    cassandraMessageHandler.setIngestQuery(cqlIngest);
+    cassandraMessageHandler.setAsync(false);
+    return cassandraMessageHandler;
+}
+----
+====
+
+The request message payload must be converted like this:
+
+====
+[source,java]
+----
+List<List<Object>> ingestBooks =
+    payload.stream()
+            .map(book ->
+                    List.<Object>of(
+                            book.isbn(),
+                            book.title(),
+                            book.author(),
+                            book.pages(),
+                            book.saleDate(),
+                            book.isInStock()))
+            .toList();
+----
+====
+
+For more sophisticated use-cases, the payload can be as an instance of `com.datastax.oss.driver.api.core.cql.Statement`.
+A `com.datastax.oss.driver.api.querybuilder.QueryBuilder` API is recommended to build various statements to execute against Apache cassandra.
+For example, to remove all the data from `Book` table, a message with a payload like this can be sent to the `CassandraMessageHandler`: `QueryBuilder.truncate("book").build()`.
+Alternatively, for a logic based on a request message, the `statementExpression` or `statementProcessor` can be provided for the `CassandraMessageHandler` to build a `Statement` based on that message.
+For convenience, a `com.datastax.oss.driver.api.querybuilder` is registered as an `import` into a SpEL evaluation context, so a target expression can be as simple as this:
```

Comment:
```suggestion
For more sophisticated use-cases, the payload can be as an instance of `com.datastax.oss.driver.api.core.cql.Statement`.
The `com.datastax.oss.driver.api.querybuilder.QueryBuilder` API is recommended to build various statements to execute against Apache Cassandra.
For example, to remove all the data from the `Book` table, a message with a payload like this can be sent to the `CassandraMessageHandler`: `QueryBuilder.truncate("book").build()`.
Alternatively, for logic based on a request message, a `statementExpression` or `statementProcessor` can be provided for the `CassandraMessageHandler` to build a `Statement` based on that message.
For convenience, a `com.datastax.oss.driver.api.querybuilder` is registered as an `import` into a SpEL evaluation context, so a target expression can be as simple as this:
```